aurelia-slickgrid/src/aurelia-slickgrid/filters/sliderFilter.ts

```diff
@@ -26,8 +26,13 @@ export class SliderFilter implements Filter {
     return this.columnDef && this.columnDef.filter && this.columnDef.filter.params || {};
   }
 
+  /** Getter for the `filter` properties */
+  private get filterProperties(): ColumnFilter {
+    return this.columnDef && this.columnDef.filter || {};
+  }
+
   get operator(): OperatorType | OperatorString {
-    return OperatorType.equal;
+    return this.columnDef.filter.operator || OperatorType.equal;
```

Comment:
@jmzagorski 
I had no choice but to code the `operator` this way. I tried to use the `this._operator` with the Getter/Setter like I saw in other filters (like the multipleSelect) but that ain't working. I might have misunderstood how it's use, but I don't see how the Setter could get called!? 

Passing the `operator` from the Column Definitions could be useful at some times (especially so for the Slider Filter, since we might want to use EQ for some grid but in other we might want '>=')

Reply:
I looked at your current file and you have the right idea. If the operator can be set by the user (e.g. it is can `EQ` or `>=` then you need to define a getter and a setter similar to the compound filters. Other filters such as the multiple select only have a getter because the filter operator should only be `in` since any other operator does not makes sense. So if the user tries to set the operator on the `MultipleSelect` it should throw a readonly error. Hopefully that helps clear that up.

Reply:
Okay i see what you mean, I was looking at the `CompoundSliderFilter`. 

For the basic `SliderFilter`, the operator should only be equal (i think) because that is the default operator and it cannot be changed by the user at run time since it is not the `CompoundSliderFilter`. So you should not need a setter, only a getter:

```typescript
  get operator(): OperatorType | OperatorString {
    return OperatorType.equal;
  }
```

Unless I misunderstood some of these filters and the programmer can change the operator on non-compound filters, and if that is the case then some of the other filters might have to be changed (e.g. InputFilter will always be `EQ`)

Reply:
@jmzagorski 
Yes I do want to provide this flexibility of passing an operator, even non-compound filters. With that in consideration, is my code correct for acquiring the `operator` or should I change something? I still don't understand what and who calls the Setter though

Reply:
i think you want it to make the `SliderCompound` setter similar to the `CompoundSliderFilter`. 

```typescript
  set operator(op: OperatorType | OperatorString) {
    this._operator = op;
  }

  get operator(): OperatorType | OperatorString {
    return this._operator || OperatorType.empty;
  }
```

the operator on the Filter interface is really just a fallback because the ColumnFilter interface operator is used first. I did this because I needed a place to put the default operator since we removed filterType from the filter service. The filter service was using a switch statement to get the default operator based on the filter type, but I had to remove this logic. Putting on the instance was practical since a class should know what it default operator should be.

Reply:
but that's the thing, I did try to use the Getter/Setter that way and it didn't work. The `this._operator` is always `undefined` even though I set it inside the grid definition as `filter: Filters.Slider, operator: '>' }` 

You might need to build the project to see the behavior :P

Reply:
i see what you are saying, The filter service never actually sets the operator. So that means you had it correctly implemented and the other filters that have a getter only should also have that same logic i guess

Or you can create a getter only (similar to the InputFilter) with whatever default operator you want because I think your filter service will use the `columnDef.filter.operator` before checking the `columnDef.filter.model.operator`

Reply:
it does work the way I coded it, it pulls the `operator` if I passed it in the grid options and if I didn't then it uses the default one. I was just wondering if that is the best code. Considering what you wrote, I guess it's the best approach, is that correct?

How do you like the slider by the way? I always thought I needed an external lib to do that but found I could do it with just CSS styling :)

Reply:
yeah I like your logic in the getter, probabaly a good idea to change the other filters that do not have that logic too.

I really do like the filter. I may try it out in my app to get user feedback

Reply:
Ok then I'm done with this PR then. I want to create 2 other Filters anyway, I realized that the input filter should be split into `number` and `float` and perhaps make the `input be an alias to `text`. I can implement the getter logic on that other PR

Also, we should take this code and make an Editor as well